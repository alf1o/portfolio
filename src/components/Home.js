import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  container: {
    height: '100%',
    textAlign: 'center',
    padding: '3vmax',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center'
  },
  topText: {
    marginBottom: '5%',
    marginTop: '-30%',
    [theme.breakpoints.up('sm')]: {
      marginTop: '-15%'
    }
  },
  bottomText: {
    marginBottom: '15%',
    [theme.breakpoints.up('sm')]: {
      marginBottom: '5%'
    }
  },
  btn: {
    margin: '0 auto'
  },
  btnText: {
    color: theme.palette.secondary.main
  }
});

/**
  Render a title and a button to start navigation.
*/
function Home({ classes, match }) {
  return (
    <Paper
      className={classes.container}
    >
      <Typography
        variant="display2"
        className={classes.topText}
      >
        Welcome
      </Typography>
      <Typography
        variant="display1"
        className={classes.bottomText}
      >
        to my portfolio site
      </Typography>
      <Button
        component={Link}
        to="/projects/Arcade Game"
        variant="raised"
        color="primary"
        classes={{ label: classes.btnText }}
        className={classes.btn}
      >
        Explore
      </Button>
    </Paper>
  );
}

Home.propTypes = {
  classes: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
};

export { Home };
export default withStyles(styles)(Home);
