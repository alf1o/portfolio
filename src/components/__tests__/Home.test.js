import React from 'react';
import Home from '../Home';
import { createShallow } from '@material-ui/core/test-utils';

describe('`Home`', () => {
  let props = {};
  let mountedHome;
  const shallow = createShallow();

  function home() {
    if (!mountedHome) mountedHome = shallow(<Home {...props} />);
    return mountedHome;
  }

  beforeEach(() => {
    props = {
      classes: {
        container: '',
        topText: '',
        bottomText: '',
        btn: '',
        btnText: ''
      },
      match: {}
    };
  });

  it('should match snapshot', () => {
    expect(home()).toMatchSnapshot();
  });
});
