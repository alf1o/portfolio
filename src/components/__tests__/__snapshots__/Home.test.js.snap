// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Home\` should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Home)
    classes={
        Object {
            "bottomText": "",
            "btn": "",
            "btnText": "",
            "container": "",
            "topText": "",
          }
    }
    match={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "bottomText": "Home-bottomText-3",
        "btn": "Home-btn-4",
        "btnText": "Home-btnText-5",
        "container": "Home-container-1",
        "topText": "Home-topText-2",
      },
      "match": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "bottomText": "Home-bottomText-3",
          "btn": "Home-btn-4",
          "btnText": "Home-btnText-5",
          "container": "Home-container-1",
          "topText": "Home-topText-2",
        },
        "match": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
